### ISSUE TITLE
createEvent message returns error "The mediaType text/plain is not supported by ObjectSerializer.parse"

### ISSUE THREAD

**Issue by kthompson23** [DataDog/datadog-api-client-typescript#392]
Running the Typescript create event example from [api/latest/events](https://docs.datadoghq.com/api/latest/events/) documentation results in the following error:

```
Error: The mediaType text/plain is not supported by ObjectSerializer.parse.
    at Function.ObjectSerializer.parse (C:\Users\kat\projects\utilities\restart-mvis-accounts\node_modules\@datadog\datadog-api-client\dist\packages\datadog-api-client-v1\models\ObjectSerializer.js:1779:15)
    at EventsApiResponseProcessor.<anonymous> (C:\Users\kat\projects\utilities\restart-mvis-accounts\node_modules\@datadog\datadog-api-client\dist\packages\datadog-api-client-v1\apis\EventsApi.js:275:51)
    at step (C:\Users\kat\projects\utilities\restart-mvis-accounts\node_modules\@datadog\datadog-api-client\dist\packages\datadog-api-client-v1\apis\EventsApi.js:46:23)
    at Object.next (C:\Users\kat\projects\utilities\restart-mvis-accounts\node_modules\@datadog\datadog-api-client\dist\packages\datadog-api-client-v1\apis\EventsApi.js:27:53)
    at fulfilled (C:\Users\kat\projects\utilities\restart-mvis-accounts\node_modules\@datadog\datadog-api-client\dist\packages\datadog-api-client-v1\apis\EventsApi.js:18:58)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

**To Reproduce**
This is the example from the documentation:
```
"use strict";
exports.__esModule = true;
process.env.DD_API_KEY = 'redacted';
process.env.DD_SITE = 'datadoghq.com';
process.env.DD_AP_KEY = 'redacted';
var datadog_api_client_1 = require("@datadog/datadog-api-client");
var configuration = datadog_api_client_1.v1.createConfiguration();
var apiInstance = new datadog_api_client_1.v1.EventsApi(configuration);
var params = {
    // EventCreateRequest | Event request object
    body: {
        aggregationKey: 'aggregationKey_example',
        alertType: 'info',
        dateHappened: 1,
        deviceName: 'deviceName_example',
        host: 'host_example',
        priority: 'normal',
        relatedEventId: 1,
        sourceTypeName: 'sourceTypeName_example',
        tags: ['environment:test'],
        text: 'Oh boy!',
        title: 'Did you hear the news today?'
    },
};
apiInstance
    .createEvent(params)
    .then(function (data) {
    console.log("API called successfully. Returned data: ".concat(JSON.stringify(data)));
})["catch"](function (error) { return console.error(error); });
```

**Environment and Versions (please complete the following information):**
- Version: 1.0.0-beta.6
- Node: v14.18.1
- OS: Windows 10

**Comment by therve** [comment]
Thank you for your report. It sounds like an issue with the backend. In the meantime, your issue is with the `dateHappened` field, you need to pass a recent timestamp to be able to submit events.

**Comment by kthompson23** [comment]
@therve I can post to the `https://api.datadoghq.com/api/v1/events` endpoint without issue though.

**Comment by therve** [comment]
WIth specifying `dateHappened: 1,` in your payload?

**Comment by therve** [comment]
FWIW I tried your example and it fails with `event too far in the past` , the issue is that the error doesn't have the proper content type so it's not parsed properly by the client. But your payload is still invalid, even if we should give a better error.

**Comment by DrPhil** [comment]
I'm on version `1.0.0-beta.6` of `@datadog/datadog-api-client` and I'm also receiving the same error: `Error: The mediaType text/plain is not supported by ObjectSerializer.parse.`. I'm getting it when I'm trying to push metrics. I'm using the typescript example from https://docs.datadoghq.com/api/latest/metrics/#submit-metrics

Node-version: v16.13.0
OS: Manjaro Linux

**Comment by kthompson23** [comment]
@therve Surprisingly yes. The returned event has the current date on it.

```
const body = {
	aggregationKey: 'aggregationKey_example',
	alertType: 'info',
	dateHappened: 1,
	deviceName: 'deviceName_example',
	host: 'host_example',
	priority: 'normal',
	relatedEventId: 1,
	sourceTypeName: 'sourceTypeName_example',
	tags: ['environment:test'],
	text: 'Oh boy!',
	title: 'Testing event post',
};
const { data } = await axios.post('https://api.datadoghq.com/api/v1/events', body, {
	headers: {
		contentType: 'application/json',
		'DD-API-KEY': process.env.DD_API_KEY,
	},
});
```

The response:
```
{
  status: 'ok',
  event: {
    id: 6269305086122181000,
    id_str: '6269305086122180270',
    title: 'Testing event post',
    text: 'Oh boy!',
    date_happened: 1637942568,
    handle: null,
    priority: 'normal',
    related_event_id: null,
    tags: [ 'environment:test' ],
    url: 'https://app.datadoghq.com/event/event?id=6269305086122180270'
  }
}
```

I understand the issue now though, thank you. In my project I was using `Date.now()` to set the `dateHappened` property. I didn't notice that the field is supposed to be seconds since the epoch and `Date.now()` is milliseconds since the epoch. By passing seconds I was able to post an event.

**Comment by DrPhil** [comment]
I think the error message could be greatly improved. In my case it was a missing environment variable. One of (DD_SITE="datadoghq.eu" DD_API_KEY="<API-KEY>" DD_APP_KEY="<APP-KEY>") was not set.

**Comment by therve** [comment]
Fixed in #399
